pkg/api/types.go

```diff
@@ -2785,6 +2802,12 @@ type DeleteOptions struct {
 	// finalizer will be added to/removed from the object's finalizers list.
 	// +optional
 	OrphanDependents *bool
+
+	// Whether and how garbage collection will be performed.
+	// Defaults to DeletePropagationDefault.
+	// Either this field or OrphanDependents may be set, but not both.
+	// +optional
+	PropagationPolicy *DeletePropagationPolicy
```

Comment:
@mbohlool when generating openapi spec, the server fails with "Failed to register open api spec for root: non-body Restful parameter type should be a simple type, but got : *v1.DeletePropagationPolicy". Why can't a query parameter has a non-simple type? Do you have suggestion on how to fix this? Thanks.